lib/analyzer/engines-analyzer.ts

```diff
@@ -0,0 +1,41 @@
+import { Docker } from '../docker';
+import { Engine } from './types';
+
+export {
+  analyze,
+};
+
+async function analyze(targetImage: string) {
+  const engines = await getEngines(targetImage);
+  return {
+    Image: targetImage,
+    AnalyzeType: 'engines',
+    Analysis: engines,
+  };
+}
+
+async function getEngines(targetImage: string) {
+  const engines: Engine[] = [];
+  const node = await getNodeEngine(targetImage);
+  if (node) { engines.push(node); }
+  return engines;
+}
+
+function getNodeEngine(targetImage: string) {
+  return new Docker(targetImage).run('node', [ '--version' ])
+    .catch(stderr => {
+      if (typeof stderr === 'string' && stderr.indexOf('not found') >= 0) {
+        return '';
+      }
+      throw new Error(stderr);
+    })
+    .then(parseNodeEngine);
+}
+
+function parseNodeEngine(version: string) {
+  const eng: Engine = {
+    name: 'node',
```

Comment:
you should add a validation for the output - it's not enough to assume that it's a valid one)
consider using a regexp for that? (x.y.z - you can remove the leading v)